File: LeetCode/Easy/1394-Find_Lucky_Integer_In_An_Array/1394-FindLuckyIntegerInAnArray.cpp
class Solution
{
public:
    int findLucky(vector<int> &arr)
    {
        std::unordered_map<int, int> freq;

        for (int num : arr)
        {
            freq[num]++;
        }

        int ans = -1;

        for (const auto &[k, v] : freq)
        {
            if (k == v && k > ans)
                ans = k;
        }

        return ans;
    }
};

File: LeetCode/Easy/1394-Find_Lucky_Integer_In_An_Array/1394-FindLuckyIntegerInAnArray.ex
defmodule Solution do
  @spec find_lucky(arr :: [integer]) :: integer
  def find_lucky(arr) do
    freq = arr
      |> Enum.reduce(%{}, fn num, acc ->
      Map.update(acc, num, 1, fn count -> count + 1 end)
    end)

    ans = -1

    Enum.reduce(Map.to_list(freq), ans, fn {k, v}, current_ans ->
      if k == v and k > current_ans do
        k
      else
        current_ans
      end
    end)
  end
end
